cli/azd/resources/starter/.azdo/pipelines/azd-up.yml

```diff
@@ -0,0 +1,45 @@
+trigger:
```

Comment:
Please consider publishing this to azure-samples instead of embedding in CLI.

You'd create a new folder under templates, called starter, and then user our CI process to publish it to azure-samples/azd-starter

Advantages:

1. You don't need to publish a new CLI version to get updated template.

Reply:
Yeah - I think that's a good idea.

Reply:
That's a great observation, and a change to the initial proposed design. I don't mind if we want to go that way. It makes our lives easier in having the capability to make changes to old released templates.

Thinking it through what independent servicing means:

For engineering: How do we version the templates, such that `azd init` with the starter selected, is always guaranteed to be valid? I think this may be a bigger problem with how our templates work in general, and maybe should be tracked separately. For example, we update our starter to take advantage of a newer feature that is released in version 1 of `azd`. User runs `azd init` on version 0. The starter fails to work.

About displaying a different name for a product:

For product: What are UX rules on when a template should and should not be rendered as a display name? We certainly don't want `azd-starter`, and now we need a display name. But should the user to run `azd init --template <display name>` and expect it to work? Would love input on the consistent UX here. Should we apply this rule across our templates?

Reply:
@wbreza Great suggestion! I created #2038 to start the template versioning discussion. There may be broader implications to think about. 

I personally think that it is okay for V1 (and maybe for foreseeable future) to couple the basic starter assets with `azd`. The basic starter assets have little meaningful constructs -- doesn't actually do meaningful that requires separate delivery servicing, i.e. less security implications. I think it's also perfectly okay from a UX perspective that you need to use the latest version of `azd` to obtain the latest up-to-date template. The reason why I would move it to a GitHub repository is to encourage OSS contributions, yet our current replication process defeats this mechanism. But I'd love any additional insights here if I'm overlooking something.

@ellismg @savannahostrowski  for final thoughts to close.

Reply:
I think that it'd be nice for us to have this not built into azd for first offering. A lot of users actually end up cloning repos and then up'ing it, completely skipping the azd init step. 

By providing the templates as part of Azure-Samples/ we make scaffolding more accessible (easier to explore as well) + meet folks in their preferred and familiar workflow (as opposed to obfuscating what a basic starter looks like until they've taken explicit azd action)

Reply:
Thanks for the UX clarification. Let's go with the repository then.

> Is having a LICENSE/README a requirement for Azure-Samples?

I believe so. Any code repositories should have LICENSE and README. Sounds like we want the same experience for git clone, so we'll add them there.

Reply:
It's moved to a template with the current change. I still think there is merit to having compiled assets that are minimal in nature. But the direction we're headed here and what I've changed is to make these look like plain-old repositories with fully expanded examples. On the dev-side, I find the authoring experience now that we have it as templates to be pretty pleasant.

I'll still keep heading down this path, and I don't have major concerns here. Will let you all know otherwise.